app/models/media_resource.rb

```diff
@@ -4,10 +4,36 @@ class MediaResource < ApplicationRecord
 
   belongs_to :episode, -> { with_deleted }, touch: true, optional: true
 
+  acts_as_paranoid
+
   enum status: [:started, :created, :processing, :complete, :error, :retrying, :cancelled]
 
   before_validation :initialize_attributes, on: :create
 
+  after_create :replace_resources!
+
+  scope :complete_or_replaced, -> do
+    with_deleted
+      .complete
+      .where("deleted_at IS NULL OR replaced_at IS NOT NULL")
+      .order("created_at DESC")
+  end
+
+  def self.build(file, position = nil)
+    media =
+      if file&.is_a?(Hash)
+        new(file)
+      elsif file&.is_a?(String)
+        new(original_url: file)
+      else
+        file
+      end
+
+    media.try(:position=, position)
```

Comment:
This code here is tied to the contents setter on the Episode right?  Does this need to be happening for all the subclasses?